test/basic/test_lightgbm.jl

```diff
@@ -0,0 +1,97 @@
+module TestLightGBM
+
+import Libdl
+using LightGBM
+using Test
+
+src_dir = abspath(joinpath(@__DIR__, "..", "..", "src"))
+
+# These set of tests use common libraries of each system to test `find_library` without having to modify env variables
+function setup_env()
+    output = Dict()
+
+    if Sys.islinux()
+        output["sample_lib"] = "libcrypt"
+    elseif Sys.isunix()
+        output["sample_lib"] = "libevent"
+    elseif Sys.iswindows()
+        output["sample_lib"] = "netmsg"
+    end
+
+    output["ref_path"] = joinpath(src_dir, "lib_lightgbm.$(Libdl.dlext)")
+    output["custom_fixture_path"] = joinpath(src_dir, "$(output["sample_lib"]).$(Libdl.dlext)")
+    output["lib_not_on_sys_fixture_path"] = joinpath(src_dir, "lib_not_on_sys.$(Libdl.dlext)")
```

Comment:
I dont get it, is this a real lib? where does this come from? why would libdl find it if it isnt linkable? If it just a funny name for lib_lightgbm.so? I can't quite figure that out from what i am seeing